### ISSUE TITLE
Missing modules when running with `--cfg loom`

### ISSUE THREAD

**Issue by ostera** [tokio-rs/tokio#5500]
Hello folks! Just ran into an issue where trying to test my code with `loom` broke in all sorts of strange ways because of missing `tokio` submodules.

Asking around and digging through sources, I learned that `fs/mod.rs` and `net/mod.rs` are explicitly disabled if the `loom` config flag is on.

I was surprised! 

I couldn't find anything in PR #1807, which seems to add this constraint, that explained why these modules are disabled.

Is this a bug? Should this just be documented elsewhere? Are there any workarounds? How do you recommend using `loom` to test something that relies on `tokio::fs/net`?

**Comment by Darksonn** [comment]
This isn't supported today. Generally, loom tests must be able to run multiple times with the exact same behavior each time, which is normally incompatible with doing IO in the test.

**Comment by ostera** [comment]
Right, that makes sense. Most of our tests in particular aren't _using_ fs/net right now, they just depend on things transitively that rely on it. So our option here is to rearchitect the app so the parts that do need fs/net are in separate crates, but that is too big an effort at the moment.

Is there room to provide an fs/net module that simply panics on use but is available at compilation time? This would at least allow folks in our position to evaluate loom without needing to do major refactors on our existing applications.
